src/sentry/tasks/relay.py

```diff
@@ -165,6 +166,94 @@ def schedule_update_config_cache(
     )
 
 
+@instrumented_task(
+    name="sentry.tasks.relay.build_project_config",
+    queue="relay_config",
+    acks_late=True,
+    soft_time_limit=TASK_SOFT_LIMIT,
+    time_limit=TASK_HARD_LIMIT,
+)
+def build_project_config(public_key=None, trigger=None, **kwargs):
+    """Build a project config and put it in the Redis cache.
+
+    This task is used to compute missing project configs, it is aggressively
+    deduplicated to avoid running duplicate computations and thus should only
+    be invoked using :func:`schedule_build_project_config`. Because of this
+    deduplication it is not suitable for re-computing a project config when
+    an option changed, use :func:`schedule_invalidate_project_config` for this.
+
+    Do not invoke this task directly, instead use :func:`schedule_build_project_config`.
+    """
+
+    try:
+        validate_args(public_key=public_key)
+
+        sentry_sdk.set_tag("public_key", public_key)
+        sentry_sdk.set_tag("update_reason", trigger)
+        sentry_sdk.set_context("kwargs", kwargs)
+
+        from sentry.models import ProjectKey
+
+        try:
+            key = ProjectKey.objects.get(public_key=public_key)
+        except ProjectKey.DoesNotExist:
+            # In this particular case, where a project key got deleted and
+            # triggered an update, we know that key doesn't exist and we want to
+            # avoid creating more tasks for it.
+            projectconfig_cache.set_many({public_key: {"disabled": True}})
+        else:
+            from sentry.models import ProjectKeyStatus
+            from sentry.relay.config import get_project_config
+
+            if key.status != ProjectKeyStatus.ACTIVE:
+                projectconfig_cache.set_many({public_key: {"disabled": True}})
+            else:
+                config = get_project_config(key.project, project_keys=[key], full_config=True)
```

Comment:
These lines are now equivalent to `config = get_projectkey_config(key)` (well, you'll have to drop `.to_dict()` on the next line) so might as well use that instead of duplicating it.